pkg/runtime/from_config.go

```diff
@@ -109,41 +118,88 @@ func (r *Runtime) newGRPC(uri string, logger *logger.Logger) *grpc.Sync {
 	}
 }
 
-func (r *Runtime) newHTTP(uri string, logger *logger.Logger) *httpSync.Sync {
+func (r *Runtime) newHTTP(config sync.ProviderConfig, logger *logger.Logger) *httpSync.Sync {
```

Comment:
Ive opted to not introduce unit tests for these contractors, although happy to if code coverage is required

Reply:
I'll leave that up to you. I'd prefer to keep the codecov happy, but if some of the additions are troublesome to tests it's fine with me. I think you've got most of the important stuff in this PR.